src/control/lib/control/pool.go

```diff
@@ -1359,9 +1351,6 @@ func GetMaxPoolSize(ctx context.Context, log logging.Logger, rpcClient UnaryInvo
 			nvmeBytes = nvmeRankBytes
 		}
 	}
-	if nvmeBytes == math.MaxUint64 {
```

Comment:
why remove this check? isn't it still needed?

Reply:
Nope, it is not needed any more as `rankNVMeFreeSpace` will contain at least one element initialized to 0 in the function https://github.com/daos-stack/daos/blob/2c02ccecc97e6b809a0f0e038394151c4184b4c0/src/control/lib/control/pool.go#L1243 
Indeed, if this map is empty, then the previous test on `scmBytes` will fail.